account_reversal/models/account_move.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2011 Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2011 Nicolas Bessi (Camptocamp)
+# Copyright 2012-2013 Guewen Baconnier (Camptocamp)
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    to_be_reversed = fields.Boolean(
+        string="To Be Reversed",
+        help="Check this box if your entry has to be reversed at the end "
+             "of period.")
+    reversal_id = fields.Many2one(
+        comodel_name='account.move', ondelete='set null', readonly=True,
+        string="Reversal Entry")
+
+    def _move_lines_reverse_prepare(self, move, date=False, journal=False,
+                                    line_prefix=False):
+        for line in move.get('line_ids', []):
+            date = date or line[2].get('date', False)
+            journal_id = journal and journal.id
+            journal_id = journal_id or line[2].get('journal_id', False)
+            name = line[2].get('name', False) or line_prefix
+            debit = line[2].get('debit', 0.)
+            credit = line[2].get('credit', 0.)
+            amount_currency = line[2].get('amount_currency', 0.)
+            if line_prefix and line_prefix != name:
+                name = ' '.join([line_prefix, name])
+            line[2].update({
+                'name': name,
+                'date': date,
+                'journal_id': journal_id,
+                'debit': credit,
+                'credit': debit,
+                'amount_currency': -amount_currency,
+            })
+        return move
+
+    def _move_reverse_prepare(self, date=False, journal=False,
+                              move_prefix=False):
+        self.ensure_one()
+        journal = journal or self.journal_id
+        if journal.company_id != self.company_id:
+            raise UserError(
+                _("Wrong company Journal is '%s' but we have '%s'") % (
+                    journal.company_id.name, self.company_id.name))
+        ref = self.ref or move_prefix
+        if move_prefix and move_prefix != ref:
+            ref = ' '.join([move_prefix, ref])
+        date = date or self.date
+        move = self.copy_data()[0]
+        move.update({
+            'journal_id': journal.id,
+            'date': date,
+            'ref': ref,
+            'to_be_reversed': False,
+            'reversal_id': self.id,
```

Comment:
I think the problem is here: to_be_reversed and reversal_id must be set on the original move, not the reversal move.